packages/sacado/test/UnitTests/Fad_Fad_KokkosTests_Cuda.cpp

```diff
@@ -43,8 +43,8 @@ int main( int argc, char* argv[] ) {
   Teuchos::GlobalMPISession mpiSession(&argc, &argv);
 
   // Initialize Cuda
-  Kokkos::InitArguments init_args;
-  init_args.device_id = 0;
+  Kokkos::InitializationSettings init_args;
+  init_args.set_device_id(0);
```

Comment:
Do you actually want to force taking the first device?
Why won't you just initialize w/o any argument and just take all the defaults?

Reply:
I'm keeping the behavior consistent with what was already there, updates are only for compatibility

Reply:
Maybe notify the developers to revisit this.  I can't really think of a good reason to do that (unless you are in a MPI context and you actually mean for all ranks to grab the 1st GPU which is kinda weird).
When you specify the GPU this way you make life of the person that runs the test harder if he wants to specify where to run (via CTest resource allocation for instance).  All he is left with is masking visible GPUs

Reply:
I have no idea why device_id = 0 is selected in these tests. Looking through the git history, it looks like it dates all the way back to when the first Sacado CUDA test file was first created in 2014 as shown [here!](https://github.com/trilinos/Trilinos/commit/d3abd5cc6d77b6005e6fc71f12ffc028709d05a4#diff-b37b8bc7163010bc9b7a115ac6dc4c304c19ad9e4ebee587a214fc3f77c8637b).  It probably had to do with early CUDA architectures/test platforms where only device 0 worked properly.  I see now reason to keep that in these tests anymore.

Reply:
Actually, the pattern of always selecting device 0 dates back to the v1 version of the original KokkosArray package in 2011 as shown [here.](https://github.com/trilinos/Trilinos/blob/0c7ddcf1ab96d196d0b5b3cf507f82dd6885cbd0/packages/kokkos/array_v1/unit_test/TestCuda.cpp)  So we can all blame Carter 😄.  This pattern was replicated as Stokhos was ported to Kokkos in 2013 and then later Sacado in 2014.  It is amazing how stuff like this sticks around for so long.

Reply:
@etphipp I like the history trace, thanks! Does dropping `init_args.set_device_id(0);` from the tests need to be done as part of this PR, or is that a separate clean up / updated?

Reply:
I can do that separately, as it isn't really related to this PR.